### PR TITLE
dev/core#1696 - Update attachment message on mailing form.

### PR DESCRIPTION
**Pull request by jitendrapurohit**
Overview
----------------------------------------
Update attachment message on mailing form.

Before
----------------------------------------
See details on https://lab.civicrm.org/dev/core/-/issues/1696

image

After
----------------------------------------
Message updated on mailing screen.

image

Comments
----------------------------------------
Gitlab  - https://lab.civicrm.org/dev/core/-/issues/1696

### CODE REVIEW

File: ang/crmAttachment/attachments.html [hunk 1]
```diff
@@ -42,5 +42,8 @@
   </div>
   <div>
     {{ts('Alternatively, you may add new files using drag/drop.')}}
+    <span ng-if="max_size">
+      {{ts('Each file must be less than %1M in size. For larger files, please upload it somewhere else and paste a link in the body of this mailing.', {1: max_size})}}
```
Comment (demeritcowboy) [inline ang/crmAttachment/attachments.html:46]: I like the idea of telling them to stick it somewhere else (grin), just not sure if the suggested wording will mean anything to the average staffer. They might think it means a custom field or activity or just not know what to do. I don't have a good suggestion though without making it really long. Maybe "... upload it to Dropbox, Google Docs, etc. and paste a link ..." That's not really much better.

File: ang/crmAttachment.js [hunk 2]
```diff
@@ -156,6 +156,11 @@
         var model = $parse(attr.crmAttachments);
         scope.att = model(scope.$parent);
         scope.ts = CRM.ts(null);
+        CRM.api3('Setting', 'getvalue', {
```
Comment (demeritcowboy) [inline ang/crmAttachment.js:159]: Is api3 still allowed? Api4 should work here:

```
CRM.api4('Setting', 'get', {
  select: ["maxFileSize"]
}).then(function(settings) {
  // do something with settings array
}, function(failure) {
  // handle failure
});
```

Comment (jitendrapurohit) [reply ang/crmAttachment.js]: Hmm, I grepped in the js/templates/ang directories and haven't seen the usage of api4 in it (except api4explorer code). As it is working, I've now replaced the same with the above code and haven't found any issues 👍

File: ang/crmAttachment/attachments.html (hunk 1 above)
Comment (jitendrapurohit) [reply ang/crmAttachment/attachments.html]: have updated the msg to the above suggestion. Thanks @demeritcowboy

**Comment by mattwire** [comment]
@jitendrapurohit @demeritcowboy At the risk of being a pain :-) I'd prefer just adding the "Each file must be less than 3M in size." That part is useful because it is factual - something will either work or won't work, depending on the server config.
The other bit, uploading somewhere else, to google drive etc. is company policy that will be different in each case.  And for some orgs uploading somewhere outside of organisation control will be a violation of their data protection policy.

**Comment by jitendrapurohit** [comment]
cool, I've simply removed the extra unwanted line :)

image

**Comment by mattwire** [comment]
Great, if @demeritcowboy confirms I will merge

**Comment by demeritcowboy** [comment]
Yes you're right it's better without the extra wording.

And now @jitendrapurohit you are a trendsetter with the first core api4 js call outside of api4 itself!

Looks good.

**Pull request merged**